### PR TITLE
Fluxify search/browse

### PR DESCRIPTION
**Pull request by jgondron**
Why: Needed to refactor this in order to fix the next/previous buttons on image zoom and in order to make the search/browse more responsive.
How: 
-Added a store that retains the search params, results, and some other properties that need to be shared between components, such as the view option. 
-Added actions that the components can use to modify the query params and rerun the query. When done, it will notify all objects that the results change. Most of this event is trickled down through re-rendering of the parent Search.jsx component, but some child components also directly listen for these events.
-Added a way to partially serialize the query params from the store and reload. This is used to recreate history navigation, so that when the user clicks forward/back, the page will reload the results using that state in the history.

### CODE REVIEW

File: app/assets/javascripts/stores/Search.js [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+      type: "GET",
+      url: url,
+      dataType: "json",
+      success: function(result) {
+        this.setItems(result.hits);
+        this.setSorts(result.sorts);
+        this.setFacets(result.facets);
+        this.emit("SearchStoreChanged", reason);
+      },
+      error: function(request, status, thrownError) {
```
Comment (jgondron) [inline app/assets/javascripts/stores/Search.js]: Hmm. Using it in the emitted event...

**Comment by jonhartzler** [comment]
:+1:

**Comment by danthewolfe** [comment]
:+1:

**Comment by danthewolfe** [comment]
Can you rename actions/Search.jsx to actions/SearchActions.jsx ?

**Comment by danthewolfe** [comment]
:+1: :shipit:

**Pull request merged**